contrib/fsl_integration/sct_plugin.py

```diff
@@ -227,33 +285,496 @@ def onButtonGM(self, event):
+
+        hbox1 = wx.BoxSizer(wx.HORIZONTAL)
+        l1 = wx.StaticText(self, wx.ID_ANY, "Vertebral Levels")
+        hbox1.Add(l1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+        self.t1 = wx.TextCtrl(self)
+        self.t1.Bind(wx.EVT_TEXT, self.OnKeyTyped)
+        hbox1.Add(self.t1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(self.rbox_contrast, 0, wx.ALL, 5)
+        sizer.Add(self.rbox_perlevel, 0, wx.ALL, 5)
+        sizer.Add(hbox1)
+        sizer.Add(button_gm, 0, wx.ALL, 5)
+        self.sizer_h.Add(sizer)
+        self.SetSizerAndFit(self.sizer_h)
+
+
+    def OnKeyTyped(self, event):
+        vert = event.GetString()
+
+
+    def onButtonCSA(self, event):
+        selected_overlay = displayCtx.getSelectedOverlay()
+        filename_path = selected_overlay.dataSource
+        contrast = self.rbox_contrast.GetStringSelection()
+        perlevel = self.rbox_perlevel.GetStringSelection()
+        vert = self.t1.GetValue()
+
+        print('Contrast:', contrast)
+
+        print('Verts:', vert)
+
+        print('Perlevel:',perlevel)
+
+        path_name = os.path.dirname(filename_path)
+        base_name = os.path.basename(filename_path)
+
+        fname, fext = base_name.split(os.extsep, 1)
+        seg_name = "{}_seg.{}".format(fname, fext)
+        segname_path = os.path.join(path_name, seg_name)
+
+        print('Tissue:', contrast)
+
+        if perlevel == 'No':
+
+            if contrast == 'GM':
+                outfilename = os.path.join(os.getcwd(), 'gmCSA.csv')
+
+            elif contrast == 'WM':
+                outfilename = os.path.join(os.getcwd(), 'wmCSA.csv')
+
+            else:
+                outfilename = os.path.join(os.getcwd(), 'totalCSA.csv')
+
+            cmd_line = "sct_process_segmentation -i {} -o {}".format(segname_path, outfilename)
+            print('Command line:', cmd_line)
+            self.call_sct_command(cmd_line)
+
+        else:
+
+            if contrast == 'GM':
+                outfilename = os.path.join(os.getcwd(), 'gmCSA_perlevel.csv')
+
+            elif contrast == 'WM':
+                outfilename = os.path.join(os.getcwd(), 'wmCSA_perlevel.csv')
+
+            else:
+                outfilename = os.path.join(os.getcwd(), 'totalCSA_perlevel.csv')
+
+            cmd_line1 = "sct_warp_template -d {} -w warp_template2anat.nii.gz".format(filename_path)
+            print('Command line:', cmd_line1)
+            self.call_sct_command(cmd_line1)
+
+            cmd_line2 = "sct_process_segmentation -i {} -vert {} -perlevel 1 -o {}".format(segname_path, vert, outfilename)
+            print('Command line:', cmd_line2)
+            self.call_sct_command(cmd_line2)
+
+#Automatically identifies the vertebral levels
+class TabPanelVertLB(SCTPanel):
+    DESCRIPTION = """This tool automatically identifies the 
+    vertebral levels. For more information, please refer to the
```

Comment:
I suggest to specify something like "You need to have both the image and the segmentation visible (click on the in eye next to the file name)"